fe/fe-core/src/main/java/org/apache/doris/planner/Planner.java

```diff
@@ -266,6 +271,57 @@ public void createPlanFragments(StatementBase statement, Analyzer analyzer, TQue
         }
     }
 
+    private class FindTuplePredicate implements com.google.common.base.Predicate<PlanNode> {
+        private final TupleId tupleId;
+
+        FindTuplePredicate(TupleId id) {
+            tupleId = id;
+        }
+
+        @Override
+        public boolean apply(PlanNode arg) {
+            return arg.getTupleIds().size() == 1 && arg.getTupleIds().get(0) == tupleId
+                    && (!arg.conjuncts.isEmpty() || !arg.runtimeFilters.isEmpty());
+        }
+    }
+
+    private boolean isNodeHasConjuncts(PlanNode rootNode, TupleId tupleId) {
+        List<PlanNode> nodes = Lists.newArrayList();
+        rootNode.collectAll(new FindTuplePredicate(tupleId), nodes);
+        return !nodes.isEmpty();
+    }
+
+    private void removeUselessRuntimeFilters(PlanNode root) {
```

Comment:
I understand that this is a very infrequently used func and should have a recognizable name.

Alternatively, we can look for other scenarios that require less RF and extend the range.

Reply:
Yes, I agree. If other scenarios are found, we can re-think how to extend this policy.